projects/common/src/utilities/angular/dynamic-component.service.ts

```diff
@@ -29,4 +34,31 @@ export class DynamicComponentService {
 
     return domPortalOutlet.attach(componentPortal);
   }
+
+  /**
+   * Dynamically create a component and insert it into the view container. Props passed
+   * to the component will be set on the component instance.
+   *
+   * @returns ComponentRef or undefined if the component could not be created
+   */
+  public attachComponentToViewContainer<T>(opts: DynamicComponentOptions<T>): ComponentRef<T> | undefined {
+    if (!isNil(opts.vcr) && !isNil(opts.component)) {
+      const componentFactory = this.componentFactoryResolver.resolveComponentFactory(opts.component);
+
+      const componentRef = opts.vcr.createComponent(componentFactory);
+      if (!isEmpty(opts.props) && !isNil(componentRef)) {
+        Object.assign(componentRef.instance, opts.props);
+      }
+
+      return componentRef;
+    }
+
+    return undefined;
+  }
+}
+
+export interface DynamicComponentOptions<T> {
```

Comment:
We should make this consistent with the existing one and generally spell things out. 
component -> componentClass
vcr -> viewContainerRef
props -> don't use props and object assignment, that's just asking for bugs (this doesn't trigger things like change detection, which is why we don't use inputs for dynamically rendered components historically). Use injector and providers.

Reply:
Updated.